src/main/java/net/pms/newgui/SharedContentTab.java

```diff
@@ -0,0 +1,748 @@
+
+		JPanel panel = builder.getPanel();
+
+		// Apply the orientation to the panel and all components in it
+		panel.applyComponentOrientation(orientation);
+
+		JScrollPane scrollPane = new JScrollPane(
+			panel,
+			JScrollPane.VERTICAL_SCROLLBAR_AS_NEEDED,
+			JScrollPane.HORIZONTAL_SCROLLBAR_AS_NEEDED
+		);
+
+		scrollPane.setBorder(BorderFactory.createEmptyBorder());
+		return scrollPane;
+	}
+
+	private PanelBuilder initSharedFoldersGuiComponents(CellConstraints cc) {
+		// Apply the orientation for the locale
+		ComponentOrientation orientation = ComponentOrientation.getOrientation(PMS.getLocale());
+		String colSpec = FormLayoutUtil.getColSpec(SHARED_FOLDER_COL_SPEC, orientation);
+
+		FormLayout layoutFolders = new FormLayout(colSpec, SHARED_FOLDER_ROW_SPEC);
+		PanelBuilder builderFolder = new PanelBuilder(layoutFolders);
+		builderFolder.opaque(true);
+
+		JComponent cmp = builderFolder.addSeparator(Messages.getString("FoldTab.7"), FormLayoutUtil.flip(cc.xyw(1, 1, 7), colSpec, orientation));
+		cmp = (JComponent) cmp.getComponent(0);
+		cmp.setFont(cmp.getFont().deriveFont(Font.BOLD));
+
+		folderTableModel = new SharedFoldersTableModel();
+		sharedFoldersList = new JTable(folderTableModel);
+		TableColumn column = sharedFoldersList.getColumnModel().getColumn(0);
+		column.setMinWidth(650);
+
+		JPopupMenu popupMenu = new JPopupMenu();
+		JMenuItem menuItemMarkPlayed = new JMenuItem(Messages.getString("FoldTab.75"));
+		JMenuItem menuItemMarkUnplayed = new JMenuItem(Messages.getString("FoldTab.76"));
+
+		menuItemMarkPlayed.addActionListener(new ActionListener() {
+			@Override
+			public void actionPerformed(ActionEvent e) {
+				String path = (String) sharedFoldersList.getValueAt(sharedFoldersList.getSelectedRow(), 0);
+				TableFilesStatus.setDirectoryFullyPlayed(path, true);
+			}
+		});
+
+		menuItemMarkUnplayed.addActionListener(new ActionListener() {
+			@Override
+			public void actionPerformed(ActionEvent e) {
+				String path = (String) sharedFoldersList.getValueAt(sharedFoldersList.getSelectedRow(), 0);
+				TableFilesStatus.setDirectoryFullyPlayed(path, false);
+			}
+		});
+
+		popupMenu.add(menuItemMarkPlayed);
+		popupMenu.add(menuItemMarkUnplayed);
+
+		sharedFoldersList.setComponentPopupMenu(popupMenu);
+		sharedFoldersList.addMouseListener(new TableMouseListener(sharedFoldersList));
+
+		/* An attempt to set the correct row height adjusted for font scaling.
+		 * It sets all rows based on the font size of cell (0, 0). The + 4 is
+		 * to allow 2 pixels above and below the text. */
+		DefaultTableCellRenderer cellRenderer = (DefaultTableCellRenderer) sharedFoldersList.getCellRenderer(0,0);
+		FontMetrics metrics = cellRenderer.getFontMetrics(cellRenderer.getFont());
+		sharedFoldersList.setRowHeight(metrics.getLeading() + metrics.getMaxAscent() + metrics.getMaxDescent() + 4);
+		sharedFoldersList.setIntercellSpacing(new Dimension(8, 2));
+
+		JImageButton but = new JImageButton("button-add-folder.png");
```

Comment:
@SubJunk should be chaged to something like "plus" image than the "folder" image.

Reply:
@valib good idea